src/amo/components/UserProfileEditPicture/index.js

```diff
@@ -19,71 +20,105 @@ type Props = {|
   user: UserType | null,
 |};
 
+type UIStateType = {|
+  hasFocus: boolean,
+|};
+
 type InternalProps = {|
   ...Props,
   i18n: I18nType,
+  setUIState: ($Shape<UIStateType>) => void,
+  uiState: UIStateType,
 |};
 
-export const UserProfileEditPictureBase = ({
-  i18n,
-  name,
-  onDelete,
-  onSelect,
-  preview,
-  user,
-}: InternalProps) => {
-  const altText = user
-    ? i18n.sprintf(i18n.gettext('Profile picture for %(name)s'), {
-        name: user.name,
-      })
-    : null;
-
-  const buttonClass = makeClassName(
-    'UserProfileEditPicture-select-button',
-    'Button Button--action Button--puffy',
-    {
-      'Button--disabled': !user,
-    },
-  );
-
-  return (
-    <section className="UserProfileEditPicture">
-      <label className="UserProfileEdit--label" htmlFor={name}>
-        {i18n.gettext('Profile photo')}
-      </label>
-
-      <UserAvatar altText={altText} preview={preview} user={user} />
-
-      {/* eslint-disable-next-line jsx-a11y/label-has-for */}
-      <label className="UserProfileEditPicture-file">
-        <input
-          accept="image/png, image/jpeg"
-          className="UserProfileEditPicture-file-input"
-          disabled={!user}
-          name={name}
-          onChange={onSelect}
-          type="file"
-        />
-        <span className={buttonClass}>{i18n.gettext('Choose Photo…')}</span>
-      </label>
-
-      {user &&
-        user.picture_url && (
-          <ConfirmButton
-            buttonType="cancel"
-            className="UserProfileEditPicture-delete-button"
-            message={i18n.gettext('Do you really want to delete this picture?')}
-            onConfirm={onDelete}
-          >
-            {i18n.gettext('Delete This Picture')}
-          </ConfirmButton>
-        )}
-    </section>
-  );
-};
-
-const UserProfileEditPicture: React.ComponentType<Props> = compose(translate())(
-  UserProfileEditPictureBase,
-);
+const initialUIState: UIStateType = { hasFocus: false };
+
+export class UserProfileEditPictureBase extends React.Component<InternalProps> {
+  onFocus = () => {
+    this.props.setUIState({ hasFocus: true });
+  };
+
+  onBlur = () => {
+    this.props.setUIState({ hasFocus: false });
+  };
+
+  render() {
+    const {
+      i18n,
+      name,
+      onDelete,
+      onSelect,
+      preview,
+      uiState,
+      user,
+    } = this.props;
+
+    const altText = user
+      ? i18n.sprintf(i18n.gettext('Profile picture for %(name)s'), {
+          name: user.name,
+        })
+      : null;
+
+    const buttonClass = makeClassName(
```

Comment:
I guess this was in the old patch too but this should be wrapped in a `Button` component. Otherwise, it makes `Button` very fragile to change since its style logic is scattered about the application. I assume there was a reason why you couldn't use the `Button` component so could you file a bug to solve that?

Reply:
yes, the reason is that it is not really a button, it is a span that looks like a button. It is a trick to style a `input[type="file"]`... I'll open an issue.

Reply:
Issue: https://github.com/mozilla/addons-frontend/issues/5623https://github.com/mozilla/addons-frontend/issues/5623